### PR TITLE
feat!: add import sorting rule

### PR DESCRIPTION
**Pull request by SgtPooki**
Lint import sorting, and support auto-fixing of import order. see https://github.com/ipfs/aegir/pull/1178 for an example

### CODE REVIEW

**Review by achingbrain — APPROVED** [review]
LGTM.  I don't love all the empty lines between groups of imports because you frequently have one lone import in a group which just looks weird sitting there all on it's own, but I don't hate them either.

I updated the title to be `feat!:` because feature will break linting for pretty much every project, though it's easy to fix.

One caveat is that `--fix` only moves the imports around - it doesn't move the accompanying `// @ts-expect-error no types` comments we have in lots of places above imports of modules that don't ship with types.

**Comment by SgtPooki** [comment]
> One caveat is that --fix only moves the imports around - it doesn't move the accompanying // @ts-expect-error no types comments we have in lots of places above imports of modules that don't ship with types.

Yea, that's important. we could update the rule to support moving associated comments, but I haven't looked into that.

File: ts.js [hunk 1]
```diff
@@ -22,6 +23,45 @@ module.exports = {
...
+          },
+          { // our relative (this package) imports
+            pattern: '.*/**',
+            group: 'index',
+            position: 'after',
+            patternOptions: { dot: true, nonegate: true }
+          }
+        ]
+      }
+    ]
```
Comment (achingbrain) [inline ts.js:65]: I think this section may be unnecessary as we extend the config from `js.js`?  At least running locally I can format `.ts` imports with the `.js` rules.

```suggestion
    'jsdoc/require-param-type': 'off' // allow compiler to derive param type
```

File: ts.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ module.exports = {
     'standard-with-typescript'
   ],
   plugins: [
-    'etc'
+    'etc',
+    'import'
```
Comment (achingbrain) [inline ts.js:15]: I think this section may be unnecessary as we extend the config from `js.js`?  At least running locally I can format `.ts` imports with the `.js` rules.

```suggestion
    'etc'
```

File: js.js [hunk 3]
```diff
@@ -72,8 +73,46 @@ module.exports = {
...
+          },
+          { // our relative (this package) imports
+            pattern: '.*/**',
+            group: 'index',
+            position: 'after',
+            patternOptions: { dot: true, nonegate: true }
+          }
+        ]
+      }
+    ]
```
Comment (achingbrain) [inline js.js:115]: How about a bit simpler?  Built-ins, externals, types then everything else.

This way the externals are sorted in the same order as the deps in `package.json`.

The only outlier here I think is "object imports" but I'm not sure we use them anywhere.

```suggestion
    'import/order': [
      'error',
      {
        alphabetize: {
          order: 'asc',
          caseInsensitive: false
        },
        'newlines-between': 'always',
        // the overall order of imports - anything not in this list is grouped together at the end
        groups: ['builtin', 'external', 'type']
      }
    ]
```

**Review by achingbrain — CHANGES_REQUESTED** [review]
A few suggestions to make this a bit simpler.

**Review by achingbrain — APPROVED** [review]

**Pull request merged**